backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/KonanDefaultParameterInjector.kt

```diff
@@ -22,7 +23,11 @@ internal class KonanDefaultParameterInjector(private val konanContext: KonanBack
             null -> IrConstImpl.constNull(startOffset, endOffset, context.irBuiltIns.nothingNType)
             PrimitiveBinaryType.BOOLEAN -> IrConstImpl.boolean(startOffset, endOffset, type, false)
             PrimitiveBinaryType.BYTE -> IrConstImpl.byte(startOffset, endOffset, type, 0)
-            PrimitiveBinaryType.SHORT -> IrConstImpl.short(startOffset, endOffset, type, 0)
+            PrimitiveBinaryType.SHORT -> when (type.getInlinedClassNative()) {
+                context.irBuiltIns.short -> IrConstImpl.short(startOffset, endOffset, type, 0)
+                context.irBuiltIns.char -> IrConstImpl.char(startOffset, endOffset, type, 0.toChar())
+                else -> error("Unexpected type with PrimitiveBinaryType.SHORT: ${type.render()}")
```

Comment:
This is fragile. What happens if type is `inline class Foo(val value: Short)`?

AFAIK, current implementation uses `Foo?` parameter type in this case, but this may change.
Please consider using `unwrapToPrimitiveOrReference` (suitable for the entire `when` too).

Reply:
Done.